src/FileCache/FileCacheBinder.cs

```diff
@@ -13,22 +13,20 @@ namespace System.Runtime.Caching
     /// <summary>
     /// You should be able to copy & paste this code into your local project to enable caching custom objects.
     /// </summary>
-    public sealed class FileCacheBinder : System.Runtime.Serialization.SerializationBinder
+    public class FileCacheBinder : System.Runtime.Serialization.SerializationBinder
```

Comment:
Regarding hashing algorithms, do you mean to recommend the built-in hash for every object?  That's what it used to use but those are no longer guaranteed to be stable across runs or platforms.

Reply:
No, GetHashCode() is not appropriate, especially not for strings. For FC you're looking for a hashing function that produces few collisions and that is decently quick. IMHO the cryptographical hashing functions included in the .NET framework such as SHA1 provide exactly that (in case of FC cryptographic security is not really relevant)
I wrote a test class, that measures the performance of the xxHash you used to SHA1.
[HashingSpeedComparison.zip](https://github.com/acarteas/FileCache/files/6964601/HashingSpeedComparison.zip)

Reply:
I'm working on another project tonight.  I'll take a look at what you have later this week.  I went with xxHash because it's supposed to be [a lot faster](https://github.com/Cyan4973/xxHash#benchmarks) than SHA1.  The C# implementation is supposed to be [at near parity to the C implementation](https://github.com/uranium62/xxHash#comparison-between-%D1%81-and-c-implementation) so I would be surprised if your testing found little difference.

Reply:
In my experience it is not that easy to compare a C/C++ benchmark results with a purely managed implementation. My test surprised even me, but I have to admin that the hash sizes differ (SHA1 = 160 bits vs. 64 for xxHash). I double-checked the AsHexString()-implementation, but it is equally performant to the one I used in the test. Also tested with 160 bits for both, but that even worsened the result for xxHash.

Reply:
I found an easily portable xxHash library at https://github.com/differentrain/YYProject.XXHash and incorporated it into the project.  Using your benchmarks, it's a little less than 2x as fast as SHA1.  `00:00:00.4326823` vs `00:00:00.2727681`.  Thanks for the all the investigation.  I'll incorporate your PR soon.

Reply:
See branch https://github.com/acarteas/FileCache/tree/f/internal_xxHash if you want to run the new version on your own machine.